plugins/engine-datafusion/jni/src/query_executor.rs

```diff
@@ -253,23 +301,102 @@ pub async fn execute_fetch_phase(
         );
     }
 
+    if(!projections.contains(&ROW_ID_FIELD_NAME.to_string())) {
+        projection_index.push(parquet_schema.index_of(ROW_ID_FIELD_NAME).unwrap());
+    }
+    projection_index.push(parquet_schema.fields.len());
+
     let file_scan_config = FileScanConfigBuilder::new(
         ObjectStoreUrl::local_filesystem(),
         parquet_schema.clone(),
         file_source,
     )
-    .with_projection(Option::from(projection_index.clone()))
+    .with_table_partition_cols(vec![Field::new(ROW_BASE_FIELD_NAME, DataType::Int64, false)])
+    .with_projection_indices(Some(projection_index.clone()))
     .with_file_group(file_group)
     .build();
 
-    let parquet_exec = DataSourceExec::from_data_source(file_scan_config);
-    let optimized_plan: Arc<dyn ExecutionPlan> = parquet_exec.clone();
+    let parquet_exec = DataSourceExec::from_data_source(file_scan_config.clone());
+
+    let projection_exprs = build_projection_exprs(file_scan_config.projected_schema())
+        .expect("Failed to build projection expressions");
+
+    let projection_exec = Arc::new(ProjectionExec::try_new(projection_exprs, parquet_exec)
+        .expect("Failed to create ProjectionExec"));
+    let optimized_plan: Arc<dyn ExecutionPlan> = projection_exec.clone();
     let task_ctx = Arc::new(TaskContext::default());
     let stream = optimized_plan.execute(0, task_ctx)?;
 
     Ok(get_cross_rt_stream(cpu_executor, stream))
 }
 
+pub fn create_projections(
+    include_fields: Vec<String>,
+    exclude_fields: Vec<String>,
+    schema: SchemaRef,
+) -> Vec<String> {
+
+    // Get all field names from schema
+    let all_fields: Vec<String> =
+        schema.fields().to_vec().iter().map(|f| f.name().to_string()).filter(|f| f.eq(ROW_ID_FIELD_NAME) || !f.starts_with("_")).collect(); //exclude metadata fields
```

Comment:
maybe need a different way to figure out the metadata fields - also lets check if user fields start with `_` ?